src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -13,912 +13,1153 @@
-            Assert.Equal(2, fullSyncedSolution.Projects.Count());
-
-            // Mutate both projects to each have a document in it.
-            solution = solution.GetProject(project1.Id).AddDocument("X.cs", SourceText.From("// X")).Project.Solution;
-            solution = solution.GetProject(project2.Id).AddDocument("Y.vb", SourceText.From("' Y")).Project.Solution;
-
-            // Now just sync project1's cone over.  We should not see the change to project2 on the remote side.
-            // But we will still see project2.
-            {
-                await solution.AppendAssetMapAsync(map, project1.Id, CancellationToken.None);
-                var project1Checksum = await solution.CompilationState.GetChecksumAsync(project1.Id, CancellationToken.None);
-                var project1SyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, project1Checksum, updatePrimaryBranch: false, CancellationToken.None);
-                Assert.Equal(2, project1SyncedSolution.Projects.Count());
-                var csharpProject = project1SyncedSolution.Projects.Single(p => p.Language == LanguageNames.CSharp);
-                var vbProject = project1SyncedSolution.Projects.Single(p => p.Language == LanguageNames.VisualBasic);
-                Assert.True(csharpProject.DocumentIds.Count == 2);
-                Assert.Empty(vbProject.DocumentIds);
-            }
-
-            // Similarly, if we sync just project2's cone over:
-            {
-                await solution.AppendAssetMapAsync(map, project2.Id, CancellationToken.None);
-                var project2Checksum = await solution.CompilationState.GetChecksumAsync(project2.Id, CancellationToken.None);
-                var project2SyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, project2Checksum, updatePrimaryBranch: false, CancellationToken.None);
-                Assert.Equal(2, project2SyncedSolution.Projects.Count());
-                var csharpProject = project2SyncedSolution.Projects.Single(p => p.Language == LanguageNames.CSharp);
-                var vbProject = project2SyncedSolution.Projects.Single(p => p.Language == LanguageNames.VisualBasic);
-                Assert.Single(csharpProject.DocumentIds);
-                Assert.Single(vbProject.DocumentIds);
-            }
-        }
+        var assetProvider = await GetAssetProviderAsync(workspace, remoteWorkspace, solution);
 
-        [Fact]
-        public async Task TestPartialProjectSync_AddP2PRef()
-        {
-            var code = @"class Test { void Method() { } }";
+        var project1Checksum = await solution.CompilationState.GetChecksumAsync(project1.Id, CancellationToken.None);
+    }
 
-            using var workspace = TestWorkspace.CreateCSharp(code);
-            using var remoteWorkspace = CreateRemoteWorkspace();
+    [Fact]
+    public void TestNoActiveDocumentSemanticModelNotCached()
```

Comment:
a bunch of tests to validate the strong vs weak holding of semantic models based on active-doc, along with tests that oop respects that.